src/node_api.cc

```diff
@@ -3554,3 +3557,299 @@ napi_status napi_run_script(napi_env env,
   *result = v8impl::JsValueFromV8LocalValue(script_result.ToLocalChecked());
   return GET_RETURN_STATUS(env);
 }
+
+struct napi_threadsafe_function__ {
+  uv_async_t async;
+  uv_idle_t idle;
```

Comment:
Could we use `Environment::SetImmediate()` instead of having to manage custom `uv_idle_t`?

Reply:
@gabrielschulhof Yeah, that makes sense. Was really asking a question there :)